bson/src/main/org/bson/codecs/configuration/mapper/ClassModel.java

```diff
@@ -34,6 +35,10 @@
     private final Map<String, FieldModel> fields = new TreeMap<String, FieldModel>();
     private final CodecRegistry registry;
     private final TypeResolver resolver;
+    private final MemberResolver memberResolver;
+    private final WeightedValue<String> collectionName = new WeightedValue<String>();
```

Comment:
I see your point but this isn't not that different than putting a field name on the FieldModel:  it's just telling the codec how/where to map this to.

As for multiple destinations, morphia, for example, allows only a single collection mapping but you can override it when calling the Datastore methods by passing the name of the collection to use.